util/ntp/ntp.go

```diff
@@ -0,0 +1,134 @@
+package ntp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/beevik/ntp"
+	"github.com/pactus-project/pactus/util/logger"
+)
+
+const (
+	maxClockOffset = time.Duration(math.MinInt64)
+)
+
+type Checker struct {
+	lock   sync.RWMutex
+	ctx    context.Context
+	cancel func()
+
+	threshold  time.Duration
+	offset     time.Duration
+	interval   time.Duration
+	logger     *logger.SubLogger
+	retryTimes int
+}
+
+func NewNtpChecker(interval time.Duration) *Checker {
+	ctxWithCancel, cancel := context.WithCancel(context.Background())
+	server := &Checker{
+		ctx:        ctxWithCancel,
+		cancel:     cancel,
+		interval:   interval,
+		threshold:  1 * time.Second,
+		retryTimes: 3,
+	}
+
+	server.logger = logger.NewSubLogger("_ntp", server)
+
+	return server
+}
+
+func (c *Checker) clockOffset() time.Duration {
+	clockOffset := maxClockOffset
+
+	for i := 0; i < c.retryTimes; i++ {
+		server := "pool.ntp.org"
+		response, err := ntp.Query(server)
+		if err != nil {
+			c.logger.Debug(
+				fmt.Sprintf("ntp error(retry %v/%v times)", i+1, c.retryTimes),
+				"server", server, "error", err,
+			)
+
+			continue
+		}
+
+		if err := response.Validate(); err != nil {
+			c.logger.Debug(
+				fmt.Sprintf("ntp error(retry %v/%v times)", i+1, c.retryTimes),
+				"server", server, "error", err,
+			)
+
+			continue
+		}
+
+		clockOffset = response.ClockOffset
+
+		break
+	}
+
+	return clockOffset
+}
+
+func (c *Checker) checkClockOffset() {
+	ticker := time.NewTicker(c.interval)
+
+	for {
+		select {
+		case <-c.ctx.Done():
+			return
+
+		case <-ticker.C:
+			c.lock.Lock()
+			c.offset = c.clockOffset()
+
+			if c.offset == maxClockOffset {
+				c.logger.Error("error on getting clock offset")
+			} else if c.OutOfSync(c.offset) {
+				c.logger.Error(
+					"The node is out of sync with the network time",
+					"threshold", c.GetThreshold(),
+					"offset", c.offset,
+				)
+			}
+			c.lock.Unlock()
+		}
+	}
+}
+
+func (c *Checker) OutOfSync(offset time.Duration) bool {
+	return math.Abs(float64(offset)) > float64(c.threshold)
+}
+
+func (*Checker) String() string {
+	return "ntp"
+}
+
+func (c *Checker) Start() {
+	go c.checkClockOffset()
+}
+
+func (c *Checker) Stop() {
+	c.cancel()
+}
+
+func (c *Checker) GetThreshold() time.Duration {
```

Comment:
I don't understand the reason for this. 
Whoever creates the NTP check also knows the threshold. Why should we report it again here?